app/mixins/search_resource.js

```diff
@@ -49,6 +50,28 @@ export default Ember.Mixin.create({
     }
   }),
 
+  onSearchCountry(searchText) {
```

Comment:
Here is a reusable mixin (well it tries to be, some of the comments I mentioned above apply here as well).

So we can't have any custom code of a controller. A mixin is not designed to stash away code, it should be a tool you integrate in the piece you're trying to build and used there to help you.

Notice how `onSearchCountry` conflicts in behaviours with `onSearchTextChange`, they have similar a purpose (triggering a search on change), but it doubles up. Which is a red flag.

@abulsayyad123 did a lot of integrations of search with this, he'll know best how to use it

Reply:
I am using a ember searchable select plugin for dropdown and it has its own **On search** method call. so can't use  **onSearchTextChange** observer.
I will create a separate mixin for this as we have to use search country at create page and update page as well.

Reply:
Your search method could trigger the onSearchTextChange. 
Although this mixing was built with infinite scroll in mind, if it's a drop-down we might not want to use it.

And again, a mixin is only appropriate if it can be re-used in multiple parts of the app. If not you can just place your methods in your controller

Reply:
I moved it in a mixin as it is used in create subform page and update subform page.